rpcs3/Emu/RSX/VK/helpers/mem_allocator.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "../VulkanAPI.h"
+#include "../../rsx_utils.h"
+#include "shared.h"
+
+#include "3rdparty/GPUOpen/include/vk_mem_alloc.h"
+
+namespace vk
+{
+	// Memory Allocator - base class
+
+	class mem_allocator_base
+	{
+	public:
+		using mem_handle_t = void *;
+
+		mem_allocator_base(VkDevice dev, VkPhysicalDevice /*pdev*/) : m_device(dev) {}
+		virtual ~mem_allocator_base() = default;
+
+		virtual void destroy() = 0;
+
+		virtual mem_handle_t alloc(u64 block_sz, u64 alignment, u32 memory_type_index) = 0;
+		virtual void free(mem_handle_t mem_handle) = 0;
+		virtual void *map(mem_handle_t mem_handle, u64 offset, u64 size) = 0;
```

Comment:
```suggestion
		virtual void* map(mem_handle_t mem_handle, u64 offset, u64 size) = 0;
```
as I've seen in any other piece of code, the `*` symbol is attached to type, and not to the name
look for `void* map` method in the class below

Reply:
[example](https://github.com/RPCS3/rpcs3/pull/9491/files#diff-0351cee9232797582bcb7436834d9e5ae1847183ddb2be12398f05cf9c023e2dR50)